examples/tutorial.dx

```diff
@@ -0,0 +1,483 @@
+  expressions are identical:
+
+:t arrSum (x2 + x2)
+
+:t arrSum $ x2 + x2
+
+' Next: `\`. This symbol is the lambda sigil in Dex. It is analogous to the
+  `lambda` keyword in Python, and starts the definition of a function value
+  (i.e. closure). In `arrSum` above: the lambda takes an argument named `acc`
+  and returns the body, which is the expression following the `.` (a `for`
+  constructor in this case).
+
+
+' That leaves: `withState`. This function uses the `State` effect, enabling us
+  to introduce imperative variables into the computation.
+  `withState` takes an initial value `init` and a body function taking a
+  "mutable value" reference (`acc` here), and returns a pair of the body
+  function's result and the final value. Here's a simple example:
+
+:p withState 10 $ \ state.
+     state := 30
+     get state
+
+' The first element of the returned pair is the body function's result (`20`).
+  The second element is the final value of the variable (`30`).
+
+' Finally: this is a good point to talk a bit about some other operators in Dex.
+  In the examples above, we see two types of equal sign operators: `=` and `:=`.
+  The first is the `let` operator that creates an immutable assignment (a
+  "let-binding"). This one is built into the language and can be used anywhere.
+
+q = for i:(Fin 10).
+        -- Bind a temporary variable `temp`, as an example.
+        temp = (ordinal i) + 10
+        for j:(Fin 5). temp
+
+' The other is `:=`, which is an assignment operator that can only be used
+  when a `State` effect is available (e.g. inside of a body function passed to
+  `withState`. `ref := x` assigns the value `x` to the mutable reference `ref`.
+  Reading the value in `ref` is possible via the `get` function. or via using
+  the final result returned by `withState`.
+
+'## Typeclasses
+
+' Our `arrSum` function is pretty neat. It lets us work with arrays with any
+  index type and computes the sum. However, `arrSum` explicitly takes only
+  integer arrays (of type `n => Int32`).
+
+:t arrSum
+
+' If we try to apply `arrSum` to float arrays, we get the following error:
+
+arrSum for i : (Fin 5). 10.0
+
+' We can compare the type of our `arrSum` function to the `sum` function found
+  in the Dex Prelude.
+
+:t sum
+
+' The Prelude-defined `sum` function also has an additional argument, spelled
+  like: `(Add v) ?=> ...`. This is a constraint telling us that the function
+  expects an `Add v` typeclass instance, where `v` is any type that implements
+  the `Add` typeclass.
+
+' We could look in the Dex Prelude to see exactly how `sum` is defined and what
+  `Add v` means. But we can guess what the `Add v` constraint  means: `v` needs
+  to be a type that works with `add`. We can do that! Let's define our own
+  typeclass.
+
+interface MyAdd a:Type
+  myAdd : a -> a -> a
+  myZero : a
+
+' This declares a typeclass (i.e. interface or trait) called `MyAdd` with some
```

Comment:
We might want to favor the word `interface` as the most important here, since this is what we use in the language. Unless you want to make the intro more geared towards Haskell people, which is fine by me too.